server/publications/collections/product.js

```diff
@@ -66,48 +58,37 @@ Meteor.publish("Product", function (productId) {
     return this.ready();
   }
   let _id;
+  const selector = {
+    isDeleted: { $in: [null, false] },
+    isVisible: true
+  };
   let productShopId;
   const shop = Reaction.getCurrentShop();
   // verify that parent shop is ready
   if (typeof shop !== "object") {
     return this.ready();
   }
 
-  // selector should come first as default, alterations take place later depending on role
-  const selector = {
-    isVisible: true,
-    isDeleted: { $in: [null, false] }
-  };
-
-  // TODO: Check with @lcampanis about why this was deleted
-  // For marketplace functionality, this seems to be a requirement,
-  // also was breaking tests.
-  if (Roles.userIsInRole(this.userId, ["owner", "admin", "createProduct"], shop._id)) {
-    selector.isVisible = {
-      $in: [true, false]
-    };
-  }
-
-
+  // Take productShopId in order to check if user can edit this product or view its revisions
   // TODO review for REGEX / DOS vulnerabilities.
   if (productId.match(/^[23456789ABCDEFGHJKLMNPQRSTWXYZabcdefghijkmnopqrstuvwxyz]{17}$/)) {
     // selector._id = productId;
     // TODO try/catch here because we can have product handle passed by such regex
     _id = productId;
-    // we need productShopId in order to check if user can edit this product or view its revisions
+
     const product = Products.findOne(_id);
     if (product) {
       productShopId = product.shopId;
     } else {
       return this.ready();
     }
   } else {
-    const newSelector = Object.assign({}, selector, {
```

Comment:
@prosf @lcampanis - this change here breaks the test that we're blocked on. It causes `newSelector` to find invisible products for non-authenticated users.

Is there a marketplace related reason for this or can I fix it by undoing this change?